Formula/y/yalantinglibs.rb

```diff
@@ -0,0 +1,49 @@
+class Yalantinglibs < Formula
+  desc "Collection of modern C++ libraries"
+  homepage "https://github.com/alibaba/yalantinglibs"
+  url "https://github.com/alibaba/yalantinglibs/archive/refs/tags/lts1.0.2.tar.gz"
+  sha256 "91049e5c2931918fe57d7513c0cd2797e07c5e642aa40da40a511b0999ae2ec7"
+  license "Apache-2.0"
+  head "https://github.com/alibaba/yalantinglibs.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "asio"
+  depends_on "async_simple"
+  depends_on "frozen"
+  depends_on "iguana"
```

Comment:
no linkage for these deps, https://github.com/Homebrew/homebrew-core/actions/runs/13746108009/job/38441201794?pr=210299#step:5:169

Reply:
No, it's not needed when building, but when using yalanting libs, these libs is needed, 
For example, see `https://github.com/alibaba/yalantinglibs/blob/main/include/ylt/struct_json/json_writer.h`
The yalantinglibs is a set of collection of header only libs including these libs, with new namespace.

Reply:
How should one handle this situation?